src/display_buffer.cc

```diff
@@ -312,7 +312,18 @@ DisplayLine parse_display_line(StringView line)
                 auto closing = std::find(it+1, end, '}');
                 if (closing == end)
                     throw runtime_error("unclosed face definition");
-                face = get_face({it+1, closing});
+                if (*(it+1) == '{' and closing+1 != end and *(closing+1) == '}')
+                {
+                    auto atom_it = atoms.find(String{it+2, closing});
+                    if (atom_it == atoms.end())
+                        throw runtime_error(format("undefined atom {}", String{it+2, closing}));
+                    for (auto display_it = atom_it->value.begin(), end = atom_it->value.end(); display_it != end; ++display_it)
```

Comment:
This is a for each, you can just write it `for (auto& atom : atom_it->value) res.push_back(atom);`